### PR TITLE
 #1447. On preInsert object in TimestampableBehavior we get different…

### PR DESCRIPTION
**Pull request by eugenekurasov**
… dataTime.

For solve this situation we need available set same time on create and update field.

### CODE REVIEW

File: src/Propel/Runtime/Util/PropelDateTime.php [hunk 1]
```diff
@@ -75,6 +80,23 @@ public static function createHighPrecision($time = null)
         return $dateTime;
     }
 
+    /**
+     * Get a DateTime object with milliseconds resolution.
+     *
+     * @param null $time
```
Comment (dereuromark) [inline src/Propel/Runtime/Util/PropelDateTime.php]: whats the alternative value? should be stated here, too.

**Comment by eugenekurasov** [comment]
Yeah, tests passed

**Comment by eugenekurasov** [comment]
@marcj can you merge this PR?
Fix for #1447

File: src/Propel/Generator/Behavior/Timestampable/TimestampableBehavior.php [hunk 2]
```diff
@@ -85,7 +85,7 @@ public function preUpdate($builder)
         if ($this->withUpdatedAt()) {
             $valueSource = strtoupper($this->getTable()->getColumn($this->getParameter('update_column'))->getType()) === 'INTEGER'
                 ? 'time()'
-                : '\\Propel\\Runtime\\Util\\PropelDateTime::createHighPrecision()'
+                : '\\Propel\\Runtime\\Util\\PropelDateTime::createSameHighPrecision(1)'
```
Comment (marcj) [inline src/Propel/Generator/Behavior/Timestampable/TimestampableBehavior.php]: createSameHighPrecision has as first argument a bool, not a integer.

File: src/Propel/Generator/Behavior/Timestampable/TimestampableBehavior.php [hunk 3]
```diff
@@ -105,20 +105,20 @@ public function preInsert($builder)
         $script = '';
 
         if ($this->withCreatedAt()) {
-            $valueSource = strtoupper($this->getTable()->getColumn($this->getParameter('create_column'))->getType()) === 'INTEGER'
+            $createdValueSource = strtoupper($this->getTable()->getColumn($this->getParameter('create_column'))->getType()) === 'INTEGER'
                 ? 'time()'
-                : '\\Propel\\Runtime\\Util\\PropelDateTime::createHighPrecision()'
+                : '\\Propel\\Runtime\\Util\\PropelDateTime::createSameHighPrecision(1)'
```
Comment (marcj) [inline src/Propel/Generator/Behavior/Timestampable/TimestampableBehavior.php]: same here

**Comment by marcj** [comment]
Can you talk a bit about that solution and why you've chosen to use a static global stored timestamp in the PropelDateTime class? I don't think that is very elegant, maybe you can show me what the actual error is so we can find a more sexy solution.

**Comment by eugenekurasov** [comment]
Hi,
I try found is more easy solution for preInsert action.
So I need to have same value for insert database on preInsert.
createHighPrecision - give me different values each time, and this is like object is was modify, but is not true.
So I need HighPrecision with save state for Insert and update. And I think **private static** is transparent solution and is small evil.

Other solution - is change generated script with additional conditions for check if created date is exist and update exist - generate once time.
But of this solution I need to check what fields type, and if they different I need to convert value.  I think it is incomprehensible solution solution.

If you have any idea how I can do it for easy understand and without static - I will be try do it.

I will be change number to boolean.

**Comment by eugenekurasov** [comment]
@marcj I set boolean type instead integer.
For preUpdate method I roll back change.

**Comment by eugenekurasov** [comment]
@marcj I have found solution without using static variable.
Can you check PR again please.

**Comment by marcj** [comment]
That looks way better, thanks!

**Pull request merged**